src/prefect/tasks/prefect/flow_run.py

```diff
@@ -106,4 +113,27 @@ def run(
 
         # grab the ID for the most recent version
         flow_id = flow[0].id
-        return client.create_flow_run(flow_id=flow_id, parameters=parameters)
+
+        # providing an idempotency key ensures that retries for this task
+        # will not create additional flow runs
+        flow_run_id = client.create_flow_run(
+            flow_id=flow_id,
+            parameters=parameters,
+            idempotency_key=context.get("flow_run_id"),
+        )
+
+        if not self.wait:
+            return flow_run_id
+
+        signals = {
+            sig._state_cls.__name__: sig for sig in PrefectStateSignal.__subclasses__()
+        }
```

Comment:
This bit feels a bit off to me - perhaps make this a utility function in the `signals` module? Something like:

```python
def from_state(state : State) -> PrefectStateSignal:
    ...
```

Reply:
Sure I can do that - to what end though?

Reply:
Just a cleanliness thing. Feels odd to dig into internal attributes like `__subclasses__` and `_state_cls` from inside the task library.

Reply:
@jcrist - updated; also added a test for the newly created function and caught a minor bug so thanks for the suggestion!